In [25]:
# importar bibliotecas e carregar o csv

import pandas as pd
import seaborn as sns

df = pd.read_csv(r'../datasets/youtube_data.csv')

In [28]:
df.head(5)

video_id                                              title  \
0  gsJAlLOFBv0                      TINY Tech That Actually Works   
1  ypicIkaiViM  AI & future of workforce: Andrew Yang on how t...   
2  1Nef8LPO-jo       5 ILLEGAL gadgets that will get you ARRESTED   
3  lCHqmzynO-s                      Overrated vs. Underrated Tech   
4  7uFrtqSwYzM  APPLE Glass Revolutionizes AR Experience Forever!   

                                         description        published_date  \
0                           No description available  2025-05-02T17:37:10Z   
1  Andrew Yang, Forward Party co-chair and former...  2025-06-18T12:39:53Z   
2  #shorts #technology \n\nI spend a LOT of time ...  2022-11-01T11:00:06Z   
3  💬 Join my Discord server: https://discord.gg/g...  2024-07-08T18:04:31Z   
4  Discover the revolutionary world of augmented ...  2024-12-22T16:49:00Z   

                 channel_id    channel_title  \
0  UCMiJRAwDNSNzuYeN2uWa0pA   Mrwhosetheboss   
1  UCrp_UI8XtuYfpiqluWLD7Lw  CNBC Television   
2  UCMiJRAwDNSNzuYeN2uWa0pA   Mrwhosetheboss   
3  UCPk2s5c4R_d-EUUNvFFODoA       Gohar Khan   
4  UCxqG_E-68WAE0TWYfIopv6Q          Digifix   

                                                tags  category_id  view_count  \
0  ['tiny', 'tech', 'gadgets', 'small', 'miniature']           28   8962092.0   
1  ['Squawk Box U.S.', 'CNBC', 'business news', '...           25    289626.0   
2                                 ['shorts', 'tech']           28  81372201.0   
3  ['thailand', 'surin', 'style', 'travel', 'day'...           27  21255964.0   
4  ['apple glasses price', 'apple glasses design'...           28   2790436.0   

   like_count  comment_count duration  \
0    243350.0          515.0    PT57S   
1      3393.0         1240.0  PT7M50S   
2   4178447.0         6378.0    PT47S   
3    909386.0         2681.0    PT31S   
4     44278.0         1359.0    PT16S   

                                        thumbnail  
0  https://i.ytimg.com/vi/gsJAlLOFBv0/default.jpg  
1  https://i.ytimg.com/vi/ypicIkaiViM/default.jpg  
2  https://i.ytimg.com/vi/1Nef8LPO-jo/default.jpg  
3  https://i.ytimg.com/vi/lCHqmzynO-s/default.jpg  
4  https://i.ytimg.com/vi/7uFrtqSwYzM/default.jpg

In [ ]:
# percebe-se que não há dados faltantes, mas há colunas que não são necessárias para a análise

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 600 entries, 0 to 599
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   video_id        600 non-null    object 
 1   title           600 non-null    object 
 2   description     600 non-null    object 
 3   published_date  600 non-null    object 
 4   channel_id      600 non-null    object 
 5   channel_title   600 non-null    object 
 6   tags            600 non-null    object 
 7   category_id     600 non-null    int64  
 8   view_count      600 non-null    float64
 9   like_count      600 non-null    float64
 10  comment_count   600 non-null    float64
 11  duration        600 non-null    object 
 12  thumbnail       600 non-null    object 
dtypes: float64(3), int64(1), object(9)
memory usage: 61.1+ KB


In [40]:
# apenas para ter certeza de que não há dados faltantes manualmente

for col in df.columns:
    if df[col].isnull().values.all() or df[col].isna().values.all():
        print(f"A coluna {col} TEM dados faltantes.")
    else:
        print(f"A coluna {col} NÃO tem dados faltantes.")

A coluna title NÃO tem dados faltantes.
A coluna description NÃO tem dados faltantes.
A coluna published_date NÃO tem dados faltantes.
A coluna channel_title NÃO tem dados faltantes.
A coluna category_id NÃO tem dados faltantes.
A coluna view_count NÃO tem dados faltantes.
A coluna like_count NÃO tem dados faltantes.
A coluna comment_count NÃO tem dados faltantes.
A coluna duration NÃO tem dados faltantes.


In [ ]:
# remover colunas que não são necessárias para a análise

df.drop(columns=['tags'], axis=1 , inplace=True)

df.head(5)

title  \
0                      TINY Tech That Actually Works   
1  AI & future of workforce: Andrew Yang on how t...   
2       5 ILLEGAL gadgets that will get you ARRESTED   
3                      Overrated vs. Underrated Tech   
4  APPLE Glass Revolutionizes AR Experience Forever!   

                                         description        published_date  \
0                           No description available  2025-05-02T17:37:10Z   
1  Andrew Yang, Forward Party co-chair and former...  2025-06-18T12:39:53Z   
2  #shorts #technology \n\nI spend a LOT of time ...  2022-11-01T11:00:06Z   
3  💬 Join my Discord server: https://discord.gg/g...  2024-07-08T18:04:31Z   
4  Discover the revolutionary world of augmented ...  2024-12-22T16:49:00Z   

     channel_title  category_id  view_count  like_count  comment_count  \
0   Mrwhosetheboss           28   8962092.0    243350.0          515.0   
1  CNBC Television           25    289626.0      3393.0         1240.0   
2   Mrwhosetheboss           28  81372201.0   4178447.0         6378.0   
3       Gohar Khan           27  21255964.0    909386.0         2681.0   
4          Digifix           28   2790436.0     44278.0         1359.0   

  duration  
0    PT57S  
1  PT7M50S  
2    PT47S  
3    PT31S  
4    PT16S

# Análise de Dados de Vídeos do YouTube 

Este notebook tem como objetivo analisar dados de desempenho de vídeos do YouTube para identificar padrões e insights.

**Q1. Qual vídeo teve mais visualizações?**
**Q2. Qual vídeo teve mais likes?**
**Q3. Qual vídeo teve mais comentários?**
**Q4. Em qual ano teve mais vídeos lançados?**


## Q1. Qual vídeo teve mais visualizações?

In [47]:
most_viewed_video = df.loc[df['view_count'].idxmax()]
print(f"O vídeo com mais visualizações é: {most_viewed_video['title']} com {int(most_viewed_video['view_count']):,} visualizações, do canal {most_viewed_video['channel_title']}.")

O vídeo com mais visualizações é: 50 Cent - Ayo Technology (Official Music Video) ft. Justin Timberlake com 343,758,991 visualizações, do canal 50CentVEVO.


## Q2. Qual vídeo teve mais likes?

In [49]:
most_liked_video = df.loc[df['like_count'].idxmax()]
print(f"O vídeo com mais likes é: {most_liked_video['title']} com {int(most_liked_video['like_count']):,} likes, do canal {most_liked_video['channel_title']}.")

O vídeo com mais likes é: New Science Project #shorts #science #technology #trending com 4,421,091 likes, do canal VMK Technical Power.


## Q3. Qual vídeo teve mais comentários?

In [50]:
most_commented_video = df.loc[df['comment_count'].idxmax()]
print(f"O vídeo com mais comentários é: {most_commented_video['title']} com {int(most_commented_video['comment_count']):,} comentários, do canal {most_commented_video['channel_title']}.")

O vídeo com mais comentários é: 50 Cent - Ayo Technology (Official Music Video) ft. Justin Timberlake com 40,241 comentários, do canal 50CentVEVO.


## Q4. Em qual ano teve mais vídeos lançados?

In [61]:
df.drop(columns=['publish_date'], inplace=True)

In [ ]:
# converter a coluna 'published_date' para o tipo datetime e extrair apenas a data

df['published_date'] = pd.to_datetime(df['published_date']).dt.date
df.head(5)

title  \
0                      TINY Tech That Actually Works   
1  AI & future of workforce: Andrew Yang on how t...   
2       5 ILLEGAL gadgets that will get you ARRESTED   
3                      Overrated vs. Underrated Tech   
4  APPLE Glass Revolutionizes AR Experience Forever!   

                                         description published_date  \
0                           No description available     2025-05-02   
1  Andrew Yang, Forward Party co-chair and former...     2025-06-18   
2  #shorts #technology \n\nI spend a LOT of time ...     2022-11-01   
3  💬 Join my Discord server: https://discord.gg/g...     2024-07-08   
4  Discover the revolutionary world of augmented ...     2024-12-22   

     channel_title  category_id  view_count  like_count  comment_count  \
0   Mrwhosetheboss           28   8962092.0    243350.0          515.0   
1  CNBC Television           25    289626.0      3393.0         1240.0   
2   Mrwhosetheboss           28  81372201.0   4178447.0         6378.0   
3       Gohar Khan           27  21255964.0    909386.0         2681.0   
4          Digifix           28   2790436.0     44278.0         1359.0   

  duration  
0    PT57S  
1  PT7M50S  
2    PT47S  
3    PT31S  
4    PT16S

In [63]:
# criar uma coluna só com o ano

df['year'] = pd.to_datetime(df['published_date']).dt.year
df.head(1)

title               description published_date  \
0  TINY Tech That Actually Works  No description available     2025-05-02   

    channel_title  category_id  view_count  like_count  comment_count  \
0  Mrwhosetheboss           28   8962092.0    243350.0          515.0   

  duration  year  
0    PT57S  2025

In [66]:
anoMaisVisto = df['year'].value_counts().sort_index()
print(f"O ano com mais vídeos lançados é: {anoMaisVisto.idxmax()} com {anoMaisVisto.max()} vídeos.")

O ano com mais vídeos lançados é: 2025 com 354 vídeos.


**Q1. Qual vídeo teve mais visualizações?**

O vídeo com mais visualizações foi o 50 Cent - Ayo Technology (Official Music Video) ft. Justin Timberlake com 343,758,991 visualizações

**Q2. Qual vídeo teve mais likes?**

New Science Project #shorts #science #technology #trending com 4,421,091 likes

**Q3. Qual vídeo teve mais comentários?**

50 Cent - Ayo Technology (Official Music Video) ft. Justin Timberlake com 40,241 comentários

**Q4. Em qual ano teve mais vídeos lançados?**

O ano com mais vídeos lançados é: 2025 com 354 vídeos